matron/src/watch.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * watch.c
+ *
+ * tracks key state, resets after timer
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <string.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <pthread.h>
+
+#include "events.h"
+
+#define WATCH_TIME 1
+
+static int count = 0;
+static int stage = 0;
+
+static pthread_t p;
+
+void *watch_time(void *);
+
+// extern def
+
+void watch_init() {
+  if (pthread_create(&p, NULL, watch_time, 0) ) {
+    fprintf(stderr, "WATCH: Error creating thread\n");
+  }
+}
+
+void watch_deinit() {
+    pthread_cancel(p);
+}
+
+void *watch_time(void *x) {
+  (void)x;
+
+  while(1) {
+    if(stage==3) count++;
+    if(count==10) {
+      fprintf(stderr, "RESTARTING...\n");
+      system("systemctl restart norns-*");
```

Comment:
yeah, pretty sure the plain `system("systemctl ...)` won't work because as soon as matron is stopped the child shell will also be killed.

this SO question has some ideas, namely:
- `system("setsid .... &")`
- something suspiciously complicated with `fork()`, `execl()`, `waitpid()` and so on